src/Infrastructure/Persistence/Initialization/DatabaseInitializer.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Finbuckle.MultiTenant;
+using FSH.WebApi.Infrastructure.Multitenancy;
+using FSH.WebApi.Shared.Multitenancy;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace FSH.WebApi.Infrastructure.Persistence.Initialization;
+
+internal class DatabaseInitializer : IDatabaseInitializer
+{
+    private readonly TenantDbContext _tenantDbContext;
+    private readonly IMultiTenantStore<FSHTenantInfo> _tenantStore;
+    private readonly DatabaseSettings _dbSettings;
+    private readonly IServiceProvider _serviceProvider;
+    private readonly ILogger<DatabaseInitializer> _logger;
+
+    public DatabaseInitializer(TenantDbContext tenantDbContext, IMultiTenantStore<FSHTenantInfo> tenantStore, IOptions<DatabaseSettings> dbSettings, IServiceProvider serviceProvider, ILogger<DatabaseInitializer> logger)
+    {
+        _tenantDbContext = tenantDbContext;
+        _tenantStore = tenantStore;
+        _dbSettings = dbSettings.Value;
+        _serviceProvider = serviceProvider;
+        _logger = logger;
+    }
+
+    public async Task InitializeDatabasesAsync(CancellationToken cancellationToken)
+    {
+        // First initialize the tenant db context
+        if (_tenantDbContext.Database.GetPendingMigrations().Any())
```

Comment:
if ((await _tenantDbContext.Database.GetPendingMigrationsAsync(cancellationToken)).Any())

Reply:
This is not an Async method / Task, hence await is not needed.

Reply:
I see many methods do not have async await decorators. Are we skipping use of await and async intentionally?

Reply:
> if ((await _tenantDbContext.Database.GetPendingMigrationsAsync(cancellationToken)).Any())

Oh now I see what you mean... unfortunately the pr is already merged... you can make one yourself for this ;-)

Reply:
> I see many methods do not have async await decorators. Are we skipping use of await and async intentionally?

When you're immediately returning a Task, it's not necessary to make the method async and then do await...
Or said in another way, when your method is async and it only uses await on the last method, it's better to take away the async and await...
(no need to set up an expensive state machine just to tear it down again)